x-pack/plugins/upgrade_assistant/public/application/components/overview/review_step/es_stats/es_stats.tsx

```diff
@@ -0,0 +1,150 @@
+import { useHistory } from 'react-router-dom';
+
+import {
+  EuiStat,
+  EuiSpacer,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiCard,
+  EuiScreenReaderOnly,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { reactRouterNavigate } from '../../../../../../../../../src/plugins/kibana_react/public';
+import { getDeprecationsUpperLimit } from '../../../../lib/utils';
+import { useAppContext } from '../../../../app_context';
+import { EsStatsErrors } from './es_stats_error';
+import { NoDeprecations } from '../no_deprecations';
+
+const i18nTexts = {
+  statsTitle: i18n.translate('xpack.upgradeAssistant.esDeprecationStats.statsTitle', {
+    defaultMessage: 'Elasticsearch',
+  }),
+  totalDeprecationsTitle: i18n.translate(
+    'xpack.upgradeAssistant.esDeprecationStats.warningDeprecationsTitle',
+    {
+      defaultMessage: 'Warning',
+    }
+  ),
+  criticalDeprecationsTitle: i18n.translate(
+    'xpack.upgradeAssistant.esDeprecationStats.criticalDeprecationsTitle',
+    {
+      defaultMessage: 'Critical',
+    }
+  ),
+  loadingText: i18n.translate('xpack.upgradeAssistant.esDeprecationStats.loadingText', {
+    defaultMessage: 'Loading Elasticsearch deprecation stats…',
+  }),
+  getCriticalDeprecationsMessage: (criticalDeprecations: number) =>
+    i18n.translate('xpack.upgradeAssistant.esDeprecationStats.criticalDeprecationsLabel', {
+      defaultMessage: 'This cluster has {criticalDeprecations} critical deprecations',
+      values: {
+        criticalDeprecations,
+      },
+    }),
+  getTotalDeprecationsMessage: (clusterCount: number, indexCount: number) =>
+    i18n.translate('xpack.upgradeAssistant.esDeprecationStats.totalDeprecationsTooltip', {
+      defaultMessage:
+        'This cluster is using {clusterCount} deprecated cluster settings and {indexCount} deprecated index settings',
+      values: {
+        clusterCount,
+        indexCount,
+      },
+    }),
+};
+
+export const ESDeprecationStats: FunctionComponent = () => {
+  const history = useHistory();
+  const { api } = useAppContext();
+
+  const { data: esDeprecations, isLoading, error } = api.useLoadUpgradeStatus();
+
+  const allDeprecations = esDeprecations?.cluster?.concat(esDeprecations?.indices) ?? [];
+  const criticalDeprecations = allDeprecations.filter(
+    (deprecation) => deprecation.level === 'critical'
+  );
+
+  const hasWarnings = allDeprecations.length > 0;
+  const hasCritical = criticalDeprecations.length > 0;
+  const hasNoDeprecations = !isLoading && !error && !hasWarnings && !hasCritical;
+  const shouldRenderStat = (forSection: boolean) => error || isLoading || forSection;
+
+  return (
+    <EuiCard
+      data-test-subj="esStatsPanel"
+      layout="horizontal"
+      title={
+        <>
+          {i18nTexts.statsTitle}
+          {error && <EsStatsErrors error={error} />}
+        </>
+      }
+      {...(!hasNoDeprecations && reactRouterNavigate(history, '/es_deprecations/cluster'))}
```

Comment:
Can we add a test to verify that clicking this navigates to the correct page? Same for the Kibana link.